app/domain/authentication/constraints/permitted_constraint.rb

```diff
@@ -0,0 +1,26 @@
+require 'command_class'
+
+module Authentication
+  module Constraints
+
+    PermittedConstraint = CommandClass.new(
+        dependencies: {
+            permitted: [],
+            logger: Rails.logger
+        },
+        inputs:   [:resource_restrictions]
+    ) do
+
+      def call
+        (@resource_restrictions.keys - @permitted).each do |not_supported_restriction_name|
+          raise Errors::Authentication::ConstraintNotSupported.new(
+              not_supported_restriction_name,
+              @permitted
+          )
+        end
```

Comment:
i think that the call method is more readable if it tells the story without logic. the logic should lie in private methods